src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const sumObject = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete newState[key];
+        });
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
```

Comment:
Consider adding a `default` case in the `switch` statement for error handling. This will ensure that if an action with an unrecognized `type` is encountered, your function can handle it gracefully, perhaps by throwing an error or ignoring the action.